Library/Homebrew/cmd/cleanup.rb

```diff
@@ -62,6 +62,10 @@ def cleanup_cache
         cleanup_path(path) { path.unlink }
         next
       end
+      if path.to_s.include?("java_cache")
```

Comment:
I'll change if if people are that in love with it, but I kind of feel like if we start trying to protect people from every funky edge case "Doh" moment in the core possible we're all going to spend the next 60 years making some very minor core tweaks.